shell/platform/android/io/flutter/plugin/platform/CurrentAccessibilityBridge.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.platform;
+
+import io.flutter.view.AccessibilityBridge;
+
+/**
+ * A package-private wrapper around the accessibility bridge.
+ *
+ * This is used to propagate an updatable reference to the accessibility bridge within the package.
+ */
+class CurrentAccessibilityBridge {
```

Comment:
Can you describe why we need this?  It looks like it's enabling switching out an instance without letting its client know that the value switched.  But I'm not sure why that's a good thing to do instead of just offering a mutator on the client and avoiding this class entirely.

Reply:
In terms of this chosen approach, I think the only recommendation I would have is that you consider calling this class `AccessibilityBridgeHolder` because the concept of something being "current" is actually a runtime statement, and ending with `AccessibilityBridge` makes this class sound like a subclass of `AccessibilityBridge`.

With regard to this approach vs others, you suggested that there might be many setter methods required to replace this class, and that many would never be used.  I'm curious why we would add setters, or even expose this class, to things that won't need it?  I'm also curious how this approach reduces exposure across the API surface.  Considering all of those classes that would have setters added to them, aren't they all being given a reference to this class, anyway?

If your analysis says that this is the most readable and maintainable option of those that exist, then maybe I'm just missing something.  But I would definitely recommend considering if this approach isn't simply re-inventing what we would normally do with a variable and a setter.  If it is, then it will probably be less readable and less maintainable over time for developers who are used to simply setting references on things.  But it's up to you.

Reply:
@amirh (optional) I thought of another potential alternative besides this, having to propagate all the set calls, or having to register callbacks. You could get rid of the `getCurrent` method on this holder class and instead wrap and expose all the `AccessibilityBridge` methods that you're expecting clients to call, so none of the clients are ever dealing with the Bridge directly and able to cache it. It would mean extra boilerplate and a maintenance burden though.

@matthew-carroll I think the root issue here is that there's a convoluted and long chain of instantiations, some of which require dependencies from upper levels. The same kind of issue that dependency injection practices and frameworks try to prevent. We need `AccessiblityBridge` constructed and maintained in `PlatformViewsController` because of onChange callbacks, but then it's used way over in `AccessibilityDelegatingFrameLayout`, which is constructed through an internal chain of `new` calls. So if setters were used  `PlatformViewsController` would need to call `VirtualDisplayController#setBridge` which would need to call `SingleViewPresentation#setBridge` which would need to call `AccessibilityDelegatingFrameLayout#setBridge` which would then be the final thing to actually use the bridge. Pub/sub systems (like [EventBus](https://github.com/google/guava/wiki/EventBusExplained) also address this type of problem, but if we want to bring in something like either that or use DI here it would be its own discussion/broader refactor.

Reply:
Went with @mklim's suggestion, made this be an `AccessibilityEventsDelegate`.

We could potentially take this a step further in making sure no-one but PlatformViewsController sets the a11y bridge by splitting this into an interface and an implementation. But I think that this extra complexity isn't justified, as this is an internal class.